src/templates/shared/bits.js

```diff
@@ -21,6 +21,18 @@ const anchor = (href, body) => html`
 
 const pageAnchor = (page) => anchor(page._path, escapeHtml(page.title));
 
+const tagAnchor = (tag, metaIndex) => {
+  const tagPage = metaIndex.findTagPageByName(tag.name);
+  return anchor(tagPage._path, tag.name);
+};
+
+const heading = (hTag, title) => html`
+  <${hTag} id="${slugify(title)}">
+    ${title}
+    <a href="#${slugify(title)}" class="header-anchor">#</a>
+  </${hTag}>
+`;
```

Comment:
Some extracted UI components to ensure we are rendering links to tag pages and headings consistently